paymail_service_provider.go

```diff
@@ -187,6 +186,25 @@ func (p *PaymailDefaultServiceProvider) RecordTransaction(ctx context.Context,
 	}, nil
 }
 
+func buildBtTx(p2pTx *paymail.P2PTransaction) *bt.Tx {
+	if p2pTx.DecodedBeef == nil {
+		res := p2pTx.DecodedBeef.GetLatestTx()
+
+		for _, input := range res.Inputs {
+			prevTxDt := find(p2pTx.DecodedBeef.Transactions, func(tx *beef.TxData) bool { return tx.Transaction.TxID() == input.PreviousTxIDStr() })
+
+			o := (*prevTxDt).Transaction.Outputs[input.PreviousTxOutIndex]
+			input.PreviousTxSatoshis = o.Satoshis
+			input.PreviousTxScript = o.LockingScript
+		}
+
+		return res
+	}
+
+	res, _ := bt.NewTxFromString(p2pTx.Hex)
```

Comment:
Can we assume that Hex will always be valid? (because the error here is skipped)

Reply:
At this point yes. We are here after some processing of received transaction or even after SPV (in BEEF case)